modules/post/linux/gather/checkvm.rb

```diff
@@ -140,19 +139,37 @@ def run
       when /vmware virtual ide|vmware pvscsi|vmware virtual platform/i
         vm = "VMware"
       when /xen_mem|xen-vbd/i
-        vm =  "Xen"
+        vm = "Xen"
       when /qemu virtual cpu version/i
         vm = "Qemu/KVM"
       when /\/dev\/vmnet/
         vm = "VMware"
       end
     end
 
+    # Check for .dockerenv file
+    if not vm
```

Comment:
You could also go with an "unless" here and on line 158 as opposed to "if not", both are cool though.